online-shop/src/components/HOCS/LoaderHOC.tsx

```diff
@@ -0,0 +1,12 @@
+import React, {Component, Props} from 'react';
+export interface LoadingProps{
+    isLoading:boolean;
+}
+
+export const withLoading = <P extends object>(Component: React.ComponentType<P>) =>
+  class WithLoading extends React.Component<P & LoadingProps> {
+    render() {
+        console.log(this.props.isLoading);
+      return  this.props.isLoading ? <div className="pageloader is-active"><span className="title">Loading</span></div>: <Component {...this.props as P} />;
```

Comment:
Typically you would not pass the loading indicator anymore to the underlying component - because it will anyway be always false when the component renders (due to the ternary condition).